### PR TITLE
docs: add example for protos

### PR DESCRIPTION
**Pull request by shreemaan-abhishek**
### Description

Ref: https://the-asf.slack.com/archives/CUC5MN17A/p1690309561460119?thread_ts=1690230991.755169&cid=CUC5MN17A

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/deployment-modes.md [hunk 1]
```diff
@@ -434,3 +434,23 @@ upstreams:
...
+      syntax = "proto3";
+      package helloworld;
+      service Greeter {
+      rpc SayHello (HelloRequest) returns (HelloReply) {}
+      }
+      message HelloRequest {
+      string name = 1;
+      }
+      message HelloReply {
+      string message = 1;
```
Comment (monkeyDluffy6017) [inline docs/en/latest/deployment-modes.md:454]: Do we need indent and blank here?

Comment (shreemaan-abhishek) [reply docs/en/latest/deployment-modes.md]: 🤦🏼‍♂️ yes, I just pushed my changes.

File: docs/en/latest/deployment-modes.md [hunk 2]
```diff
@@ -434,3 +434,24 @@ upstreams:
...
+      service Greeter {
+        rpc SayHello (HelloRequest) returns (HelloReply) {}
+      }
+      message HelloRequest {
+        string name = 1;
+      }
+      message HelloReply {
+        string message = 1;
+      }
+```
```
Comment (nfrankel) [inline docs/en/latest/deployment-modes.md:457]: Please don't forget `#END`

Comment (shreemaan-abhishek) [reply docs/en/latest/deployment-modes.md:458]: Good catch.

**Review by nfrankel — APPROVED** [review]

**Review by monkeyDluffy6017 — APPROVED** [review]

**Pull request merged**